sdk/nodejs/resource.ts

```diff
@@ -389,6 +389,28 @@ export abstract class Resource {
         return this.__providers[pkg];
     }
 
+    /**
+     * The type assigned to the resource at construction.
+     */
+    public getResourceType(): string {
```

Comment:
In Java at least we went for `pulumiResourceType` and `pulumiResourceName` I think, which I think have even less risk of clashing and e.g. in Python mean you can do `pulumi_resource_name` (with the `get_`), which _may_ be preferable? (Equally, you _might_ feel like making them properties is OK then, but either way...)

Reply:
Yeh dotnet has the win that if it's a method it _definitely_ won't clash with property names (although as we add Call support on resources, that could still clash names, but then probably not arguments).
There's less point in making these methods in python/node because they could still clash.

Just checked java and saw that it _also_ used to have getResourceType but it got renamed. Wonder if we should rename dotnet to match? Just call this `pulumiResourceType` everywhere, making it a property where we can and leave Go and Java as just methods called `pulumiResourceType`?